tools/c7n_mailer/c7n_mailer/smtp_delivery.py

```diff
@@ -0,0 +1,21 @@
+import smtplib
```

Comment:
Add headers

Reply:
this is a refactor re headers.. i'm a little unclear what the value add on the refactor was. we dont want to multiply providers X transports. transports should be provider independent and reusable across providers. the only minor wrinkle is encrypted secrets, but the extant use here previously is safe.

Reply:
i'd suggest we just move this back into email delivery module, since its only use is with email.

Reply:
`email_delivery.py` module has a ton of AWS specific code. I wanted to specifically extract provider independent code, however some aspects (secret encryption) can be provider specific, so I would prefer to keep it this way. 

I don't think that a single `kms_decrypt` method that will be able to handle every encryption is the right solution.

Reply:
A suggestion is to create an abstract `class SecretManager` with a `decrypt` method and make a concrete class for `KMSSecretManager` and `KeyVaultSecretManager`, then we can have different implementations for each secret vault. When we instantiate the the EmailDelivery class we can inject the correct SecretManager.

Reply:
Just added `decrypt` function that will redirect to a provider specific implementation.

Reply:
@logachev thought about a bit more and how to decouple email delivery, and this seems like a good way to go. I'd like to avoid subclasses for the same transport on a provider specific basis, so the deligation on decrypt sounds fine. longer term i'd like to move aws ses to its own transport in the aws module, and have email delivery take transport in its constructor to use. the smtp transport/delivery class will need some reworking if its going to represent long lived connections (aka handle reconnects).

Reply:
To your point, @kapilt : `email_delivery` is a mess now.. There is a lot of code that can be shared across providers (like render an email\build 'to:' list etc).. So yeah, extracting ses sounds like a good idea. Ideally, I think we need a shared queue processor with some plugins from different providers.

Reply:
the mess extends to all of mailer tbh.. but incremental steps to cleaning it up seem like the best path, although i'm tempted by alternatives (🔥&  🏗 ).. i know @thisisshi had a branch that tried to do a lot of incremental cleanups, but i think he shelved it..

Reply:
fwiw, that refactor was reopened its in #3519